app/models/authentication.rb

```diff
@@ -118,6 +118,15 @@ def self.build_credential_options
     end
   end
 
+  def self.class_from_request_data(data)
+    if !data.key?('type') || data['type'] == to_s
+      return self
+    end
+    type = descendants.find { |klass| klass.name == data['type'] }
+    raise _('Must be an Authentication type') unless type
+    type
+  end
```

Comment:
1. I don't like this on the model here.  It feels like it's a generic thing that's being put into a single model
2. This feels like a duplication of a lot of the work that the NewWithTypeStiMixin does.  That is, finding the right subclass to instantiate via type.  Given that, I feel like the "type" logic belongs to create_in_provider_queue itself, somehow at the base class, in a similar manner to how NewWithTypeStiMixin works.